Mathlib/CategoryTheory/Generator.lean

```diff
@@ -567,4 +569,98 @@ theorem wellPowered_of_isDetector [HasPullbacks C] (G : C) (hG : IsDetector G) :
   haveI := small_subsingleton ({G} : Set C)
   wellPowered_of_isDetecting hG
 
+theorem wellPowered_of_isSeparator [HasPullbacks C] [Balanced C] (G : C) (hG : IsSeparator G) :
+    WellPowered C := wellPowered_of_isDetecting hG.isDetector
+
+section HasGenerator
+
+variable (E : Type u₃) [Category.{v₃} E]
```

Comment:
It might be clearer to the reader to enclose the class definitions in a `section` that has `variable (C)`, otherthey they (like me) might wonder why you have to introduce a new category.